packages/Main/src/Layer/VpcLayer.js

```diff
@@ -0,0 +1,341 @@
+    elt.obj.boxHelper.material.linewidth = 2;
+    elt.obj.boxHelper.frustumCulled = false;
+    elt.obj.boxHelper.position.copy(elt.tightbbox.min).add(box3.max);
+    elt.obj.boxHelper.autoUpdateMatrix = false;
+    layer.bboxes.add(elt.obj.boxHelper);
+    elt.obj.boxHelper.updateMatrix();
+    elt.obj.boxHelper.updateMatrixWorld();
+}
+
+function computeSSEPerspective(context, pointSize, spacing, elt, distance) {
+    if (distance <= 0) {
+        return Infinity;
+    }
+    const pointSpacing = spacing / 2 ** elt.depth;
+    // Estimate the onscreen distance between 2 points
+    const onScreenSpacing = context.camera.preSSE * pointSpacing / distance;
+    // [  P1  ]--------------[   P2   ]
+    //     <--------------------->      = pointsSpacing (in world coordinates)
+    //                                  ~ onScreenSpacing (in pixels)
+    // <------>                         = pointSize (in pixels)
+    return Math.max(0.0, onScreenSpacing - pointSize);
+}
+
+function computeSSEOrthographic(context, pointSize, spacing, elt) {
+    const pointSpacing = spacing / 2 ** elt.depth;
+
+    // Given an identity view matrix, project pointSpacing from world space to
+    // clip space. v' = vVP = vP
+    const v = new THREE.Vector4(pointSpacing);
+    v.applyMatrix4(context.camera.camera3D.projectionMatrix);
+
+    // We map v' to the screen space and calculate the distance to the origin.
+    const dx = v.x * 0.5 * context.camera.width;
+    const dy = v.y * 0.5 * context.camera.height;
+    const distance = Math.sqrt(dx * dx + dy * dy);
+
+    return Math.max(0.0, distance - pointSize);
+}
+
+function computeScreenSpaceError(context, pointSize, spacing, elt, distance) {
+    if (context.camera.camera3D.isOrthographicCamera) {
+        return computeSSEOrthographic(context, pointSize, spacing, elt);
+    }
+
+    return computeSSEPerspective(context, pointSize, spacing, elt, distance);
+}
+
+function markForDeletion(elt) {
+    if (elt.obj) {
+        elt.obj.visible = false;
+        if (__DEBUG__) {
+            if (elt.obj.boxHelper) {
+                elt.obj.boxHelper.visible = false;
+            }
+        }
+    }
+
+    if (!elt.notVisibleSince) {
+        elt.notVisibleSince = Date.now();
+        // Set .sse to an invalid value
+        elt.sse = -1;
+    }
+    for (const child of elt.children) {
+        markForDeletion(child);
+    }
+}
+
+/**
+ * A layer for [Cloud Optimised Point Cloud](https://copc.io) (COPC) datasets.
+ * See {@link PointCloudLayer} class for documentation on base properties.
+ *
+ * @extends {PointCloudLayer}
+ *
+ * @example
+ * // Create a new VPC layer
+ * const vpcSource = new VpcSource({
+ *     url: 'https://data.geopf.fr/chunk/telechargement/download/lidarhd_fxx_ept/vpc/index.vpc',
+ * });
+ *
+ * const vpcLayer = new VpcLayer('VPC', {
+ *     source: vpcSource,
+ * });
+ *
+ * View.prototype.addLayer.call(view, vpcLayer);
+ */
+class VpcLayer extends PointCloudLayer {
```

Comment:
Can you also document class properties please ? (this.roots, this.spacing, etc.)